### PR TITLE
many small changes

### LINKED ISSUE #29: add test for `roc run main.roc -- examples build`

### PR DESCRIPTION
**Pull request by Anton-4**
- renamed folders as temporary workaround for https://github.com/roc-lang/roc/issues/5430
- nix setup
- css changes
- slimmed down amount of text
- additional CI test, closes #29

### CODE REVIEW

File: examples/index.md [hunk 1]
```diff
@@ -1,25 +1,15 @@
...
-If you have any ideas or improvements for this repository please start a discussion over on [Roc Zulip chat](https://roc.zulipchat.com/).
\ No newline at end of file
+- [Json](/JsonBasic/README.html)
+- [Random Numbers](/RandomNrs/README.html)
+- [Pattern Matching](/PattMatch/README.html)
+- [Graph Traversal](/GraphTrav/README.html)
+- [Parser](/Parser/README.html)
+- [Arithmetic](/Arithmetic/README.html)
+- [Least Square Difference](/LeastSqrs/README.html)
+- [Towers of Hanoi](/TowersOfH/README.html)
```
Comment (Anton-4) [inline examples/index.md:15]: I changed the order a bit here.

File: examples/index.md [hunk 2]
```diff
@@ -1,25 +1,15 @@
...
-<!-- TODO reenable when roc-lang/examples/issues/24 is fixed -->
-<!-- - [Json](/JsonBasic/README.html) -->
-
-## Feedback
-
-Please log any issues on [GitHub](https://github.com/roc-lang/examples/issues/new). 
-
-## Ideas and Improvements
-
-If you have any ideas or improvements for this repository please start a discussion over on [Roc Zulip chat](https://roc.zulipchat.com/).
```
Comment (Anton-4) [inline examples/index.md:25]: I created an equivalent shorter text in the header.

File: flake.nix [hunk 3]
```diff
@@ -6,25 +6,42 @@
     #nixpkgs.url = "github:nixos/nixpkgs/nixos-22.05";
     # to easily make configs for multiple architectures
     flake-utils.url = "github:numtide/flake-utils";
+    # rust is used for the static-site-gen platform
+    # rust from nixpkgs has some libc problems, this is patched in the rust-overlay
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
```
Comment (Anton-4) [inline flake.nix:14]: To be able to use the static-site-gen platform

**Comment by Anton-4** [comment]
The tests are failing because of the folder name changes, once I upload tomorrow's nightly I can go back to the original folder names, so I'm going to finish this up after.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -26,6 +26,9 @@ jobs:
       - name: Get the archive from the url.
         run: curl -OL ${{ env.ROC_RELEASE_URL }}
 
+      - name: store short commit SHA
+        run: echo "SHORT_COMMIT_SHA=$(ls | grep roc_nightly | cut -d '-' -f 6 | cut -d '.' -f 1)"  >> $GITHUB_ENV
```
Comment (Anton-4) [inline .github/workflows/tests.yml:30]: To be able to get necessary dependencies like static-site-gen with the same commit as the nightly.

File: .gitignore [hunk 5]
```diff
@@ -1,6 +1,11 @@
...
 example
-examples/JsonBasic/json-basic
-examples/ParserBasic/parser-basic
+examples/Arithmetic/arithmetic
+examples/FizzBuzz/fizz-buzz
+examples/HelloWorld/hello-world
+examples/Json/json-basic
+examples/Parser/parser-basic
+examples/RandomNumbers/random-numbers
+examples/TowersOfHanoi/towers-of-hanoi
```
Comment (Anton-4) [inline .gitignore:10]: ignore the executables

File: main.roc [hunk 6]
```diff
@@ -1,7 +1,7 @@
 app "generate-build"
-    packages { pf: "/Users/luke/Documents/GitHub/roc/examples/static-site-gen/platform/main.roc" }
+    packages { pf: "/home/runner/work/examples/examples/roc_nightly/examples/static-site-gen/platform/main.roc" }
```
Comment (Anton-4) [inline main.roc:2]: this is the path used by the CI machine

**Comment by Anton-4** [comment]
Can you review @lukewilliamboswell? I am not yet able to ask you as reviewer in the normal way until you accept your invite for write permission for this repo.

File: main.roc (hunk 6 above)
Comment (lukewilliamboswell) [reply main.roc:2]: Ok, we should update README specifically in `## To view the examples site locally` section it still says to use `roc run main.roc -- examples build` which doesn't work with this change.

File: main.roc [hunk 7]
```diff
@@ -36,18 +36,15 @@ view = \fileName, htmlContent ->
...
-                    a [href "https://roc.zulipchat.com/"] [text "Feedback"]
-                ],
-                li [] [
-                    a [href "https://github.com/roc-lang/examples"] [text "Contribute"]
-                ]
+            p [class "feedback"] [
+                text "we 💜 feedback: ",
+                a [href "https://github.com/roc-lang/examples"] [text "GitHub"],
+                text " | ",
+                a [href "https://roc.zulipchat.com/"] [text "Group Chat"],
```
Comment (lukewilliamboswell) [inline main.roc:43]: Love the new feedback paragraph 😄

**Review by lukewilliamboswell — APPROVED** [review]
The changes look good. CI scripts run well for me locally. I like the css updates.

**Comment by fdbeirao** [comment]
There's a small typo in README.md: the path to the example is now `JsonBasic` instead of `json-basic`. Perhaps that can be added to this PR?

Where it reads `roc run examples/json-basic/main.roc` it should read `roc run examples/JsonBbasic/main.roc`

Cheers

**Comment by Anton-4** [comment]
Thanks @lukewilliamboswell and @fdbeirao! I've made changes according to your feedback.

**Review by fdbeirao — APPROVED** [review]
Thanks for renaming JsonBasic to Json and updating the README.md 🙏🏻

**Pull request merged**